src/middlewared/middlewared/plugins/pool.py

```diff
@@ -619,6 +620,49 @@ def transform(dataset):
 
         verrors = ValidationErrors()
         await self.__common_validation(verrors, 'pool_dataset_create', data, 'CREATE')
+
+        if '/' not in data['name']:
+            verrors.add('pool_dataset_create.name', 'You need a full name, e.g. pool/newdataset')
+        else:
+            parent = await self.middleware.call(
+                'zfs.dataset.query',
+                [('id', '=', data['name'].split('/')[0])]
+            )
+
+            if not parent:
+                verrors.add(
+                    'pool_dataset_create.name',
+                    'Please specify a pool which exists for the dataset/volume to be created'
+                )
+            else:
+                parent = parent[0]
+
+                if data['type'] == 'VOLUME' and 'volsize' in data:
+
+                    avail_mem = int(parent['properties']['available']['rawvalue'])
+
+                    if (
+                        data['volsize'] > (avail_mem * 0.80) and
```

Comment:
You must also do this validation on update, making that not the right math, you must discount the current used size of the zvol.